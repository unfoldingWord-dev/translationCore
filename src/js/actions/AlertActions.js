import types from './ActionTypes';

export const closeAlert = (id) => ({
  type: types.CLOSE_ALERT,
  id,
});

/**
 * Opens a new alert.
 * @param {string} id - the alert id
 * @param {*} message - the message to display in the alert
 * @param {func} onConfirm - custom confirm handler
 * @param {func} onCancel - custom cancel handler
 * @param {string} confirmText - the confirm button text
 * @param {string} cancelText - the cancel button text
 * @return {*}
 */
export const openAlert = (id, message, {
  onConfirm = null,
  onCancel = null,
  confirmText = null,
  cancelText = null,
} = {}) => ({
  type: types.OPEN_ALERT,
  id,
  message,
  onConfirm,
  confirmText,
  onCancel,
  cancelText,
});

/**
 * Opens a new alert that can be ignored.
 * @param {string} id - the alert id
 * @param {*} message
 * @param {func} onConfirm - custom confirm handler
 * @param {func} onCancel - custom cancel handler
 * @param {func} onIgnore - custom ignore handler
 * @param {string} confirmText - the confirm button text
 * @param {string} cancelText - the cancel button text
 * @param {string} ignoreText - the ignore checkbox text
 * @return {*}
 */
export const openIgnorableAlert = (id, message, {
  onConfirm = null,
  onCancel = null,
  onIgnore = null,
  confirmText = null,
  cancelText = null,
  ignoreText = null,
} = {}) => ({
  type: types.OPEN_ALERT,
  id,
  message,
  onConfirm,
  confirmText,
  onCancel,
  cancelText,
  onIgnore: typeof onIgnore === 'function' ? onIgnore : () => {},
  ignoreText,
});

/**
 * Ignores a specific alert for the rest of the session
 * @param {string} id - the alert id to ignore
 * @param {boolean} ignore - ignores or enables the alert
 * @return {*}
 */
export const ignoreAlert = (id, ignore = true) => ({
  type: types.IGNORE_ALERT,
  id,
  ignore,
});
