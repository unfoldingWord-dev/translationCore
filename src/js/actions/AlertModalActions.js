/**
 * @module Actions/AlertModal
 */

import consts from './ActionTypes';

/**
 * @description opens the alert dialog with the specified alert message.
 * @param {String} alertMessage - message to be displayed inside the alert dialog.
 * @param {Boolean} loading - true displays spinning icon and no action button in dialog.
 * @return {object} action content.
 */
export function openAlertDialog(alertMessage, loading) {
  return {
    type: consts.OPEN_ALERT_DIALOG,
    alertMessage,
    loading,
  };
}

/**
 * @description opens the option dialog with the specified alert message, and options.
 * @param {String} alertMessage - message to be displayed inside the alert dialog.
 * @param {function} callback - callback function that is called when clicking on button.  First parameter is text on button. Exception is when callback2 is given.
 * @param {String} button1Text - button text to show right button.
 * @param {String} button2Text - button text to show left of right button. (optional - if not present button is not added)
 * @param {String} buttonLinkText - button text to show on left link button. (optional - if not present button is not added)
 * @param {function} callback2 - optional callback function that is called user clicks on button2Text.
 * @return {Object} action content.
 */
export function openOptionDialog(alertMessage, callback, button1Text, button2Text, buttonLinkText = null, callback2 = null) {
  return {
    type: consts.OPEN_OPTION_DIALOG,
    alertMessage,
    callback,
    callback2,
    button1Text,
    button2Text,
    buttonLinkText,
  };
}

/**
 * @description closes the alert dialog.
 * @return {object} action content.
 */
export function closeAlertDialog() {
  return { type: consts.CLOSE_ALERT_DIALOG };
}
