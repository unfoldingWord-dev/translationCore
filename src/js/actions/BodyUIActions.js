import {
  getHomeScreenStep,
  getActiveHomeScreenSteps
} from '../selectors';
import types from './ActionTypes';

/**
 * @description toggles the home view based on param.
 * @param {boolean} boolean - true or false either shows or hides it.
 */
export const toggleHomeView = boolean => ({
  type: types.TOGGLE_HOME_VIEW,
  boolean
});

export const toggleWelcomeSplash = () => ({
  type: types.TOGGLE_WELCOME_SPLASH
});

export const goToNextStep = () => {
  return ((dispatch, getState) => {
    const stepIndex = getHomeScreenStep(getState());
    dispatch(goToStep(stepIndex + 1));
  });
};

export const goToPrevStep = () => {
  return ((dispatch, getState) => {
    const stepIndex = getHomeScreenStep(getState());
    dispatch(goToStep(stepIndex - 1));
  });
};

/**
 * Goes to specified step
 * @param {number} stepNumber - Number of step to go to in home stepper
 */
export const goToStep = stepNumber => {
  return ((dispatch, getState) => {
    if (stepNumber >= 0 && stepNumber <= 3) {
      const activeSteps = getActiveHomeScreenSteps(getState());
      if (!activeSteps[stepNumber]) return;
      dispatch({
        type: types.GO_TO_STEP,
        stepIndex: stepNumber
      });
    } else if (stepNumber < 0) {
      console.error('The min number of steps is 0. (0-3)');
    } else {
      console.error('The max number of steps is 3. (0-3)');
    }
  });
};

export const toggleProjectsFAB = () => ({
  type: types.TOGGLE_PROJECTS_FAB
});

export const openOnlineImportModal = () => ({
  type: types.OPEN_ONLINE_IMPORT_MODAL
});

export const closeOnlineImportModal = () => {
  return {
    type: types.CLOSE_ONLINE_IMPORT_MODAL
  };
};

export const openLicenseModal = () => ({
  type: types.OPEN_LICENSE_MODAL
});

export const closeLicenseModal = () => ({
  type: types.CLOSE_LICENSE_MODAL
});

/**
 * @description show or not show dimmed screen.
 * @param {bool} enable dims the screen if enabled otherwise removes the dim
 */
export const dimScreen = (enable=true) => ({
  type: types.SHOW_DIMMED_SCREEN,
  bool: enable
});
