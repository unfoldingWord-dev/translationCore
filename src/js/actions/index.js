/**
 * @author Ian Hoegen
 * @description: This serves as the new store and events api.
 * @usage:
 * Actions.sendAction('YOUR_EVENT', data) // Emits an event of type YOUR_EVENT, adds data to store
 * Actions.getData('YOUR_EVENT') // Retrieves data from the event emitted
 ***********************************************************************************/

const EventEmitter = require('events').EventEmitter;
const Reducers = require('../reducers');

class Actions extends EventEmitter {
  constructor() {
    super();
    this.store = [];
  }
  sendAction(eventName, actionObj) {
    if(Reducers[eventName]) {
      this.store[eventName] = Reducers[eventName](this.store[eventName], actionObj);
    } else {
      this.store[eventName] = actionObj;
    }
    this.emitChange(eventName);
  }

  subscribe(eventName, callback) {
    this.on(eventName, callback);
  }

  unsubscribe(eventName, callback) {
    this.removeListener(eventName, callback);
  }

  emitChange(eventName, data) {
    this.emit(eventName, data);
  }

  getData (actionType) {
    return this.store[actionType];
  }
}

var actions = new Actions();
module.exports = actions;
