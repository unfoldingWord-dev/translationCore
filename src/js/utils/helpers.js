import Path from 'path';
import * as fs from 'fs-extra';
import ManifestGenerator from '../components/core/create_project/ProjectManifest';
import BooksOfBible from '../components/core/BooksOfBible';

module.exports.isOldTestament = function (projectBook) {
    var passedBook = false;
    for (var book in BooksOfBible) {
        if (book == projectBook) passedBook = true;
        if (BooksOfBible[book] == "Malachi" && passedBook) {
            return true;
        }
    }
    return false;
}

module.exports.checkIfValidBetaProject = function (manifest) {
    if (manifest && manifest.project) return manifest.project.id == "eph" || manifest.project.id == "tit";
    else if (manifest && manifest.ts_project) return manifest.ts_project.id == "eph" || manifest.ts_project.id == "tit";
}

/**
 * 
 * @param {*} path 
 * @param {*} file 
 */
export function loadFile(path, file) {
    try {
        var manifest = fs.readJsonSync(Path.join(path, file));
        return manifest;
    }
    catch (e) {
        return null
    }
}

/**
 * @description - Generates and saves a translationCore manifest file
 * @param {string} projectSaveLocation - Filepath of where the translationCore manifest file will
 * be saved. Must be an ABSOLUTE PATH
 * @param {object} data - The translationCore manifest data to be saved
 * @param {object} tsManifest - The translationStudio manifest data loaded from a translation
 * studio project
 */
module.exports.saveManifest = function (projectSaveLocation, link, oldManifest, currentUser) {
    var data = {
        user: [currentUser],
        repo: link
    }
    var manifest;
    try {
        var manifestLocation = Path.join(projectSaveLocation, 'manifest.json');
        if (oldManifest.package_version == '3') {
            //some older versions of ts-manifest have to be tweaked to work
            manifest = this.fixManifestVerThree(oldManifest);
        } else {
            manifest = ManifestGenerator(data, oldManifest);
        }
        fs.outputJsonSync(manifestLocation, manifest);
    }
    catch (err) {
        console.error(err);
    }
    return manifest;
}

/**
 * @desription - Uses the tc-standard format for projects to make package_version 3 compatible
 * @param oldManifest - The name of an employee.
 */
module.exports.fixManifestVerThree = function (oldManifest) {
    var newManifest = {};
    try {
        for (var oldElements in oldManifest) {
            newManifest[oldElements] = oldManifest[oldElements];
        }
        newManifest.finished_chunks = oldManifest.finished_frames;
        newManifest.ts_project = {};
        newManifest.ts_project.id = oldManifest.project_id;
        newManifest.ts_project.name = this.convertToFullBookName(oldManifest.project_id);
        for (var el in oldManifest.source_translations) {
            newManifest.source_translations = oldManifest.source_translations[el];
            var parameters = el.split("-");
            newManifest.source_translations.language_id = parameters[1];
            newManifest.source_translations.resource_id = parameters[2];
            break;
        }
    } catch (e) {
        console.error(e);
    }
    return newManifest;
}

module.exports.convertToFullBookName = function (bookAbbr) {
    if (!bookAbbr) return;
    return BooksOfBible[bookAbbr.toString().toLowerCase()];
}