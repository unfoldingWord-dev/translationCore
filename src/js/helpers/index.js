import { generateTimestamp } from './TimestampGenerator.js';
import * as loadHelpers from './LoadHelpers.js';

export {
  generateTimestamp,
  loadHelpers,
};
