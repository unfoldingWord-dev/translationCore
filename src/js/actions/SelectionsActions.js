import types from './ActionTypes';
import isEqual from 'deep-equal';
// actions
import * as AlertModalActions from './AlertModalActions';
// helpers
import {generateTimestamp} from '../helpers/index';
import {checkSelectionOccurrences} from '../helpers/selectionHelpers';
import {getTranslate, getUsername, getSelections} from '../selectors';
import * as gatewayLanguageHelpers from '../helpers/gatewayLanguageHelpers';

/**
 * This method adds a selection array to the selections reducer.
 * @param {Array} selections - An array of selections.
 * @param {String} userName - The username of the author of the selection.
 * @param {Boolean} invalidated - if true then selection if flagged as invalidated, otherwise it is not flagged as invalidated
 * @param {Object} contextId - optional contextId to use, otherwise will use current
 * @return {Object} - An action object, consisting of a timestamp, action type,
 *                    a selection array, and a username.
 */
export const changeSelections = (selections, userName, invalidated = false, contextId = null) => {
  return ((dispatch, getState) => {
    contextId = contextId || getState().contextIdReducer.contextId; // use current if contextId is not passed
    const {
      gatewayLanguageCode,
      gatewayLanguageQuote
    } = gatewayLanguageHelpers.getGatewayLanguageCodeAndQuote(getState());

    dispatch({
      type: types.CHANGE_SELECTIONS,
      modifiedTimestamp: generateTimestamp(),
      gatewayLanguageCode,
      gatewayLanguageQuote,
      selections,
      userName
    });
    dispatch({
      type: types.TOGGLE_SELECTIONS_IN_GROUPDATA,
      contextId,
      selections
    });
    dispatch({
      type: types.SET_INVALIDATION_IN_GROUPDATA,
      contextId,
      invalidated
    });
  });
};
/**
 * @description This method validates the current selections to see if they are still valid.
 * @param {String} targetVerse - target bible verse.
 * @return {Object} - dispatches the changeSelections action.
 */
export function validateSelections(targetVerse) {
  return (dispatch, getState) => {
    const username = getUsername(getState());
    const selections = getSelections(getState());
    const validSelections = checkSelectionOccurrences(targetVerse, selections);
    const results = {
      selectionsChanged: (selections.length !== validSelections.length)
    };
    if (results.selectionsChanged) {
      dispatch(changeSelections(validSelections, username, true));
    }

    dispatch(validateAllSelectionsForVerse(targetVerse, results));
    if (results.selectionsChanged) {
      const translate = getTranslate(getState());
      dispatch(AlertModalActions.openAlertDialog(translate('tools.selections_invalidated')));
    }
  };
}

export const validateAllSelectionsForVerse = (targetVerse, results) => {
  return (dispatch, getState) => {
    const state = getState();
    const username = getUsername(state);
    const contextId = state.contextIdReducer.contextId;
    const groupsDataForVerse = getGroupDataForVerse(state, contextId);

    for (let groupItemKey of Object.keys(groupsDataForVerse)) {
      const groupItem = groupsDataForVerse[groupItemKey];
      for (let occurrenceKey of Object.keys(groupItem)) {
        const checkingOccurrence = groupItem[occurrenceKey];
        const selections = checkingOccurrence.selections;
        if (selections && selections.length) {
          const validSelections = checkSelectionOccurrences(targetVerse, selections);
          if (selections.length !== validSelections.length) {
            results.selectionsChanged = true;
            dispatch(changeSelections(validSelections, username, true, checkingOccurrence.contextId));
          }
        }
      }
    }
  };
};

/**
 * @description gets the group data for the current verse from groupsDataReducer
 * @param {Object} state
 * @param {Object} contextId
 * @return {object} group data object.
 */
export const getGroupDataForVerse = (state, contextId) => {
  const  { groupsData } = state.groupsDataReducer;
  const filteredGroupData = {};
  for (let groupItemKey of Object.keys(groupsData)) {
    const groupItem = groupsData[groupItemKey];
    for (let occurrenceKey of Object.keys(groupItem)) {
      const checkingOccurrence = groupItem[occurrenceKey];
      if (isEqual(checkingOccurrence.contextId.reference, contextId.reference)) {
        if (!filteredGroupData[groupItemKey]) {
          filteredGroupData[groupItemKey] = [];
        }
        filteredGroupData[groupItemKey].push(checkingOccurrence);
      }
    }
  }
  return filteredGroupData;
};

