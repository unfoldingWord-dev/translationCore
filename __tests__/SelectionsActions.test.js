import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import path from 'path-extra';
import fs from "fs-extra";
// actions
import {generateTimestamp} from "../src/js/helpers";
import * as SelectionsActions from '../src/js/actions/SelectionsActions';
// constants
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const PROJECTS_PATH = path.join(__dirname, 'fixtures', 'checkData');

jest.unmock('fs-extra');

describe('SelectionsActions.validateAllSelectionsForVerse', () => {
  const bookId = 'tit';

  it('No selection changes', () => {
    // given
    const targetVerse =  "Paul, a servant of God and an apostle of Jesus Christ, for the faith of God's chosen people and the knowledge of the truth that agrees with godliness, ";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId);
    const results = {
      selectionsChanged: false
    };

    // when
    store.dispatch(SelectionsActions.validateAllSelectionsForVerse(targetVerse, results));

    // then
    const actions = store.getActions();
    expect(actions.length).toEqual(0);
  });

  it('apostle selection edited', () => {
    // given
    const targetVerse =  "Paul, a servant of God and an apostl2 of Jesus Christ, for the faith of God's chosen people and the knowledge of the truth that agrees with godliness, ";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId);
    const results = {
      selectionsChanged: false
    };

    // when
    store.dispatch(SelectionsActions.validateAllSelectionsForVerse(targetVerse, results));

    // then
    const actions = store.getActions();
    expect(cleanOutDates(actions)).toMatchSnapshot();
  });

  it('all selections edited', () => {
    // given
    const targetVerse =  "";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId);
    const results = {
      selectionsChanged: false
    };

    // when
    store.dispatch(SelectionsActions.validateAllSelectionsForVerse(targetVerse, results));

    // then
    const actions = store.getActions();
    expect(cleanOutDates(actions)).toMatchSnapshot();
  });
});

describe('SelectionsActions.validateSelections', () => {
  const bookId = 'tit';
  const selectionsReducer = {
    gatewayLanguageCode: "en",
    gatewayLanguageQuote: "authority, authorities",
    "selections": [
      {
        "text": "apostle",
        "occurrence": 1,
        "occurrences": 1
      }
    ],
    username: 'dummy-test',
    modifiedTimestamp: generateTimestamp()
  };

  it('No selection changes', () => {
    // given
    const targetVerse = "Paul, a servant of God and an apostle of Jesus Christ, for the faith of God's chosen people and the knowledge of the truth that agrees with godliness, ";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId, selectionsReducer);

    // when
    store.dispatch(SelectionsActions.validateSelections(targetVerse));

    // then
    const actions = store.getActions();
    expect(actions.length).toEqual(0);
  });

  it('apostle selection edited', () => {
    // given
    const targetVerse =  "Paul, a servant of God and an apostl2 of Jesus Christ, for the faith of God's chosen people and the knowledge of the truth that agrees with godliness, ";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId, selectionsReducer);

    // when
    store.dispatch(SelectionsActions.validateSelections(targetVerse));

    // then
    const actions = store.getActions();
    expect(cleanOutDates(actions)).toMatchSnapshot();
  });

  it('god selection edited in different context', () => {
    // given
    const targetVerse =  "Paul, a servant of Go and an apostle of Jesus Christ, for the faith of God's chosen people and the knowledge of the truth that agrees with godliness, ";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId, selectionsReducer);

    // when
    store.dispatch(SelectionsActions.validateSelections(targetVerse));

    // then
    const actions = store.getActions();
    expect(cleanOutDates(actions)).toMatchSnapshot();
  });

  it('all selections edited', () => {
    // given
    const targetVerse =  "";
    const projectPath = path.join(PROJECTS_PATH, 'en_tit');
    const store = initGroupData(projectPath, bookId, selectionsReducer);

    // when
    store.dispatch(SelectionsActions.validateSelections(targetVerse));

    // then
    const actions = store.getActions();
    expect(cleanOutDates(actions)).toMatchSnapshot();
  });
});

//
// helpers
//

function cleanOutDates(actions) {
  const cleanedActions = JSON.parse(JSON.stringify(actions));
  for (let action of cleanedActions) {
    if (action.modifiedTimestamp) {
      delete action.modifiedTimestamp;
    }
  }
  return cleanedActions;
}

function initGroupData(projectPath, bookId, selectionsReducer) {
  const contextId = {
    reference: {
      bookId: bookId,
      chapter:1,
      verse: 1
    },
    groupId: 'apostle'
  };
  const groupsDataReducer = fs.readJSONSync(path.join(projectPath, 'groupsDataReducer.json'));
  const groupsIndexReducer = fs.readJSONSync(path.join(projectPath, 'groupsIndexReducer.json'));

  const initialState = {
    actions: {},
    groupsDataReducer,
    groupsIndexReducer,
    toolsReducer: {
      currentToolName: 'translationWords'
    },
    loginReducer: {
      loggedInUser: false,
      userdata: {
        username: 'dummy-test'
      }
    },
    projectDetailsReducer: {
      manifest: {
        project: {
          id: bookId
        }
      },
      projectSaveLocation: path.resolve(projectPath),
      currentProjectToolsSelectedGL: {
        translationWords: 'en'
      }
    },
    contextIdReducer: {
      contextId
    }
  };
  if (selectionsReducer) {
    initialState.selectionsReducer = selectionsReducer;
  }
  return mockStore(initialState);
}

