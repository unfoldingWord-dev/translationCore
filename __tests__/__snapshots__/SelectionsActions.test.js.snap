// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionsActions.validateAllSelectionsForVerse all selections edited 1`] = `
Array [
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "god",
      "occurrence": 1,
      "quote": "Θεοῦ",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G23160",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "god",
      "occurrence": 1,
      "quote": "Θεοῦ",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G23160",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
]
`;

exports[`SelectionsActions.validateAllSelectionsForVerse apostle selection edited 1`] = `
Array [
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
]
`;

exports[`SelectionsActions.validateSelections all selections edited 1`] = `
Array [
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "god",
      "occurrence": 1,
      "quote": "Θεοῦ",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G23160",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "god",
      "occurrence": 1,
      "quote": "Θεοῦ",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G23160",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "alertMessage": "tools.selections_invalidated",
    "type": "OPEN_ALERT_DIALOG",
  },
]
`;

exports[`SelectionsActions.validateSelections apostle selection edited 1`] = `
Array [
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "apostle",
      "occurrence": 1,
      "quote": "ἀπόστολος",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G06520",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "alertMessage": "tools.selections_invalidated",
    "type": "OPEN_ALERT_DIALOG",
  },
]
`;

exports[`SelectionsActions.validateSelections god selection edited in different context 1`] = `
Array [
  Object {
    "gatewayLanguageCode": "en",
    "gatewayLanguageQuote": "apostle, apostles, apostleship",
    "selections": Array [],
    "type": "CHANGE_SELECTIONS",
    "userName": "dummy-test",
  },
  Object {
    "contextId": Object {
      "groupId": "god",
      "occurrence": 1,
      "quote": "Θεοῦ",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G23160",
      ],
      "tool": "translationWords",
    },
    "selections": Array [],
    "type": "TOGGLE_SELECTIONS_IN_GROUPDATA",
  },
  Object {
    "contextId": Object {
      "groupId": "god",
      "occurrence": 1,
      "quote": "Θεοῦ",
      "reference": Object {
        "bookId": "tit",
        "chapter": 1,
        "verse": 1,
      },
      "strong": Array [
        "G23160",
      ],
      "tool": "translationWords",
    },
    "invalidated": true,
    "type": "SET_INVALIDATION_IN_GROUPDATA",
  },
  Object {
    "alertMessage": "tools.selections_invalidated",
    "type": "OPEN_ALERT_DIALOG",
  },
]
`;
